### DISCUSSION TITLE
3D run by using NEMO output

### DISCUSSION THREAD

**Discussion by Novaoceans** [Parcels-code/Parcels#1413] [General]
Hello everyone, I have a question on the 3D module.

We are trying to use 3D velocity fields from a NEMO model. The model uses the z-level coordinate system with bottom partial steps and nonlinear free surface is implemented using the variable volume level (VVL) scheme, which allows the stretching and compression of the thickness of vertical layers according to the changes of sea surface height. How should we use such model output with OceanParcels 3D module?

We noticed NEMO w is the velocities relative to the grid and ignores the movement of the levels, e.g., w at the free surface is always zero. Do we need to bring in the time dependent depth and add the movement of the levels to the w variable? Or does OceanParcels expect velocities relative to the moving grid?

Could you please give us some suggestions and examples on how to deal with the w output from NEMO with a time and space dependent vertical coordinate? The tutorial on the moving grid only includes u and v velocities. Thanks!

**Comment by erikvansebille** [comment]
Hi @Novaoceans, thanks for your question and sorry for the delay in responding. It's summer here ;-)

As for the surface layer: I am not aware of anyone who has already used the NEMO-VVL scheme in Parcels. While support for time-varying depth layers is implemented (see https://docs.oceanparcels.org/en/latest/examples/tutorial_timevaryingdepthdimensions.html), so far we haven't used it. You could give it a try?

As for the bottom layer: we have discussed bottom partial steps with @cpierard before, and came up with an idea how to deal with it, but haven't implemented or tested it yet. It will require writing a custom kernel for the advection, and having an extra Field that gives the partial cell for each horizontal grid cell (which is likely already in the mesh file?)
